spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteTablePathSparkAction.java

```diff
@@ -728,4 +724,22 @@ private String getMetadataLocation(Table tbl) {
         !metadataDir.isEmpty(), "Failed to get the metadata file root directory");
     return metadataDir;
   }
+
+  @VisibleForTesting
+  Broadcast<Table> tableBroadcast() {
+    if (tblBroadcast == null) {
+      this.tblBroadcast = sparkContext().broadcast(SerializableTableWithSize.copyOf(table));
+    }
+
+    return tblBroadcast;
+  }
+
+  @VisibleForTesting
+  Broadcast<Map<Integer, PartitionSpec>> specsBroadcast(Map<Integer, PartitionSpec> specsById) {
```

Comment:
Why broadcast this separately rather than using the specs [map from the `SerializableTable`](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/SerializableTable.java#L208)?

Reply:
Agreed, I think the original implementation had 2 broadcasts but there's really no need for it as far as I can tell.

Reply:
Ah, how could I have missed that? Let me combine them.